source/Camera/Basic/C++Halcon/CaptureViaZivid/CaptureViaZivid.cpp

```diff
@@ -19,50 +19,68 @@ HalconCpp::HObjectModel3D zividToHalconPointCloud(const Zivid::PointCloud &point
 	const auto width = pointCloud.width();
 	const auto height = pointCloud.height();
 
-	std::vector<float> pointsX(width * height);
-	std::vector<float> pointsY(width * height);
-	std::vector<float> pointsZ(width * height);
-	std::vector<float> colorsR(width * height);
-	std::vector<float> colorsG(width * height);
-	std::vector<float> colorsB(width * height);
-	std::vector<int> rowMapping(width*height);
-	std::vector<int> colMapping(width*height);
-
-	int numValidPoints = 0;
-	for (size_t i = 0; i < width*height; ++i)
+
+
+	int numberOfValidPoints = std::count_if(pointCloud.dataPtr(), pointCloud.dataPtr() + pointCloud.size(), [](Zivid::Point point)
+	{
+		return (!point.isNaN());
+	});
+
+	HalconCpp::HTuple tupleX, tupleY, tupleZ, tupleR, tupleB, tupleG, tupleRow, tupleCol, tupleMapping;
+	tupleX[numberOfValidPoints - 1] = (float) 0.0;
+	tupleY[numberOfValidPoints - 1] = (float) 0.0;
+	tupleZ[numberOfValidPoints - 1] = (float) 0.0;
+	tupleR[numberOfValidPoints - 1] = (Hlong)0;
+	tupleG[numberOfValidPoints - 1] = (Hlong)0;
+	tupleB[numberOfValidPoints - 1] = (Hlong)0;
+
+	tupleMapping[2 * numberOfValidPoints + 2 - 1] = (Hlong)0;
+	tupleMapping[0] = (Hlong)width;
+	tupleMapping[1] = (Hlong)height;
+
+	double* arrayX = tupleX.DArr();
+	double* arrayY = tupleY.DArr();
+	double* arrayZ = tupleZ.DArr();
+	Hlong* arrayR = tupleR.LArr();
+	Hlong* arrayG = tupleG.LArr();
+	Hlong* arrayB = tupleB.LArr();
+	Hlong* arrayMapping = tupleMapping.LArr();
+
+	int validPointIndex = 0;
+
+	auto data = pointCloud.dataPtr();
```

Comment:
can we call this pointCloudData? Just data is a bit generic.

Reply:
done